applications: nrf_desktop: Add cpu_meas, profiler_sync, smp doc

MarekPieta opened the pull request:
Change adds documentation for the following modules: cpu_meas, profiler_sync, smp.

MarekPieta commented on applications/nrf_desktop/doc/smp.rst:
```diff
@@ -33,12 +33,12 @@ Make sure that the following Kconfig options are enabled:
   See the :ref:`zephyr:device_mgmt` for details about Zephyr's device management subsystem.
 * :option:`CONFIG_MCUMGR_SMP_BT`
   Device receives SMP commands over Bluetooth.
-* ``CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_ENABLE``
```
The `CONFIG_DESKTOP_...` Kconfig options are specific to nrf_desktop project.
Their documentation pages are not generated in the documentation. Therefore its not possible to refer to these options using `:option:`. We have to leave them the with the previous syntax.

MarekPieta replied:
I will revert this part of your change

FrancescoSer replied:
Thanks for the note, fixed

MarekPieta commented on applications/nrf_desktop/doc/cpu_meas.rst:
```diff
@@ -18,15 +18,15 @@ Module events
-This Kconfig option selects the :option:`CONFIG_CPU_LOAD`.
-:option:`CONFIG_CPU_LOAD` enables the :ref:`cpu_load` that is used to perform the measurements.
+Enable the module using the ``CONFIG_DESKTOP_CPU_MEAS_ENABLE`` Kconfig option.
+This Kconfig option selects the ``CONFIG_CPU_LOAD`` option.
```
Changed to :option:`CONFIG_CPU_LOAD` since this option is not specific to nrf_desktop.

MarekPieta commented on applications/nrf_desktop/doc/profiler_sync.rst:
```diff
@@ -23,24 +23,24 @@ Module events
-Predefined signal on the GPIO is used to simultaneously generate synchronization profiler events on both devices.
-For this reason, enable the :option:`CONFIG_GPIO`.
+A predefined signal on the GPIO is used to simultaneously generate synchronization profiler events on both devices.
+For this reason, you must enable the ``CONFIG_GPIO`` option.
```
This Kconfig option is used by Zephyr. We need to use :option: syntax

MarekPieta commented on applications/nrf_desktop/doc/profiler_sync.rst:
```diff
@@ -3,12 +3,12 @@
 The profiler data is collected separately from both devices using dedicated Python scripts.
-Then the data can be merged together using the ``merge_data.py`` script with ``sync_event`` used as synchronization event for both peripheral and central.
-For more detailed information, see :ref:`profiler` documentation.
+The data can then be merged using the ``merge_data.py`` script with ``sync_event`` used as a synchronization event for both peripheral and central.
```
Used ``Peripheral`` and ``Central`` to be consistent in the module documentation page.

MarekPieta commented on applications/nrf_desktop/doc/smp.rst:
```diff
@@ -28,27 +28,28 @@ Configuration
 
 The module periodically submits the ``ble_smp_transfer_event`` while the image is being uploaded.
-This event is used by :ref:`nrf_desktop_ble_latency` that decreases the connection slave latency to speed up the transfer.
+This event is used by :ref:`nrf_desktop_ble_latency` to start decreasing the connection slave latency to speed up the transfer.
```
It's more like: `ble_latency` module is informed that the slave latency needs to be decreased when it receives this event. I rephrased this.

MarekPieta commented on applications/nrf_desktop/doc/smp.rst:
```diff
@@ -28,27 +28,28 @@ Configuration
-Modifying the data in the secondary image slot by both modules would result in broken firmware image that would be rejected by the bootloader.
+If the image is being received through the SMP, the Simple Management Protocol application module acquires the lock.
+Also, firmware update through the :ref:`nrf_desktop_dfu` is disabled until the nRF Desktop device is rebooted.
+Any modification of the data in the secondary image slot by both modules would result in a broken firmware image that would be rejected by the bootloader.
```
I would prefer not to use word `any` here, since it could be a bit misleading.

MarekPieta commented on applications/nrf_desktop/doc/smp.rst:
```diff
@@ -3,10 +3,10 @@
 
 The DFU can be performed among others using the nRF Connect Android application.
-The ``DFU`` button appears in the Bluetooth connected device's tab on Android device.
+The ``DFU`` button appears in the *Bluetooth connected devices* tab on the Android device.
```
I meant that the button is visible in the tab representing given device. Rephrased,

MarekPieta commented on applications/nrf_desktop/doc/dfu.rst:
```diff
@@ -3,7 +3,7 @@
 ##############################
 
-The DFU module is used for obtaining the update image from a transport and storing it in the appropriate partition on the flash memory.
+Use the DFU module to obtain the update image from a transport channel and to store it in the appropriate partition on the flash memory.
```
I think the we should stick to `transport` since its more general

MarekPieta commented on applications/nrf_desktop/doc/dfu.rst:
```diff
@@ -18,115 +18,116 @@ Module events
-The lock must also be taken for FLASH area erase operation.
-This is done to make sure that only one application module could modify secondary image FLASH partition.
-Modifying the data by multiple application modules would result in broken image that would be rejected by the bootloader.
+During the transfer operation, the ``dfu_lock``, located at :file:`src/util/dfu_lock.h`, must be taken by the application module.
```
The mentioned header file describes the API, the lock is actually "located" in the `dfu_lock.c` file. Can we leave it with parenthesis ?

MarekPieta commented on applications/nrf_desktop/doc/dfu.rst:
```diff
@@ -18,115 +18,116 @@ Module events
-This is done to make sure that only one application module could modify secondary image FLASH partition.
-Modifying the data by multiple application modules would result in broken image that would be rejected by the bootloader.
+During the transfer operation, the ``dfu_lock``, located at :file:`src/util/dfu_lock.h`, must be taken by the application module.
+The lock must also be taken for any flash area erase operation.
```
changed to `a flash area erase operation`

MarekPieta commented:
I separated commit related to `dfu` module.
I squashed the commit with review and added github comments to mark places where I made additional changes.
I also updated configuration channel related part of dfu module documentation (separate commit that will be squashed).

MarekPieta commented:
Applied the fixes

FrancescoSer reviewed (approved):
Seems fine to me
